modules/delegates.js

```diff
@@ -316,11 +316,15 @@ __private.checkDelegates = function (publicKey, votes, state, cb) {
 			}
 
 			if (math === '+' && (delegates != null && delegates.indexOf(publicKey) !== -1)) {
-				return setImmediate(cb, 'Failed to add vote, account has already voted for this delegate');
+				return modules.accounts.getAccount({publicKey: publicKey, isDelegate: 1 }, function (err, account) {
+					return setImmediate(cb, `Failed to add vote, delegate '${account.username}' already voted for`);
```

Comment:
What if we get an error from `getAccount` and `account` will be an `undefined`?

Reply:
I figured the fact that delegates came from the database this would be validated at an earlier step. IE if a public key is sent that already exists in account.delegates it must also have an account, let me know

Reply:
this check may need to be implemented in the if statement that checks to remove votes